### ISSUE TITLE
Add Twitter logo to geoportal header or footer

### ISSUE THREAD

**Issue by karenmajewicz** [BTAA-Geospatial-Data-Project/geoportal#345]

**Comment by Ziiiiing** [comment]
My issue:
---------
When I went through the [Local Development Quickstart]() from Wiki to get setup for Geoportal development, the `bundle install` statement came the following issue:

```console
$ bundle install
Fetching gem metadata from https://rubygems.org/............
Your bundle is locked to mimemagic (0.3.5), but that version could not be found
in any of the sources listed in your Gemfile. If you haven't changed sources,
that means the author of mimemagic (0.3.5) has removed it. You'll need to update
your bundle to a version other than mimemagic (0.3.5) that hasn't been removed
in order to install.
```

Checking all versions of the [mimemagic](https://rubygems.org/gems/mimemagic/versions) from [https://rubygems.org/](https://rubygems.org/), I found the mimemagic (0.3.5) has been yanked and were no longer supported.



My solution:
------------
update both gem and dependencies:
```console
$ bundle update mimemagic
```

**Comment by ewlarson** [comment]
Hi @Ziiiiing - Rails and mimemagic had a now famous [licensing dependency issue](https://twitter.com/nateberkopec/status/1374722404228853762) last week. We'll want to upgrade our Rails dependency to resolve that issue.

**Comment by mberkowski** [comment]
I missed this issue last week - I'll update the develop branch today.

**Comment by ewlarson** [comment]
Hey @mberkowski // I think it'll be a bit harder than that to remove mimemagic. We'll also need to update the GBL Sidecar Images plugin to remove its dependency on v0.3 of mimemagic. I opened a ticket here:
https://github.com/geoblacklight/geoblacklight_sidecar_images/issues/25

I'm going to do some work for the B1G this week and next week, so I'll try and get that done ASAP.

**Comment by mberkowski** [comment]
@ewlarson I hadn't looked too hard yet but I thought they just relicensed it back to MIT in 0.3.10. If it is that simple I can just drop it in, but if it's in your pipeline otherwise I can leave it.

**Comment by Ziiiiing** [comment]
Hello @ewlarson @mberkowski , thanks for your reply.

I am not professional in Rails, but I have tried to run `bundle update mimemagic` first so that the `bundle install` succeeded with no errors anymore. However, when I executed the following `bundle exec rake geoportal:server` statement, a new error came out. See details below:

```console
$ bundle update mimemagic
Fetching gem metadata from https://rubygems.org/............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies....
Using rake 12.3.3
Using concurrent-ruby 1.1.7
Using i18n 1.8.5
Using minitest 5.14.2
Using thread_safe 0.3.6
Using tzinfo 1.2.7
Using activesupport 5.2.4.4
Using builder 3.2.4
Using erubi 1.9.0
Using mini_portile2 2.4.0
Using nokogiri 1.10.10
Using rails-dom-testing 2.0.3
Using crass 1.0.6
Using loofah 2.7.0
Using rails-html-sanitizer 1.3.0
Using actionview 5.2.4.4
Using rack 2.2.3
Using rack-test 1.1.0
Using actionpack 5.2.4.4
Using nio4r 2.5.4
Using websocket-extensions 0.1.5
Using websocket-driver 0.7.3
Using actioncable 5.2.4.4
Using globalid 0.4.2
Using activejob 5.2.4.4
Using mini_mime 1.0.2
Using mail 2.7.1
Using actionmailer 5.2.4.4
Using activemodel 5.2.4.4
Using arel 9.0.0
Using activerecord 5.2.4.4
Using mimemagic 0.3.10 (was 0.3.5)
Using marcel 0.3.3
Using activestorage 5.2.4.4
Using public_suffix 3.1.1
Using addressable 2.5.2
Using ansi 1.5.0
Using execjs 2.7.0
Using autoprefixer-rails 9.6.1
Using awesome_print 1.8.0
Using bcrypt 3.1.13
Using bindex 0.8.1
Using deprecation 1.0.0
Using jbuilder 2.9.1
Using kaminari-core 1.2.1
Using kaminari-actionview 1.2.1
Using kaminari-activerecord 1.2.1
Using kaminari 1.2.1
Using bundler 2.1.4
Using method_source 1.0.0
Using thor 0.19.4
Using railties 5.2.4.4
Using sprockets 3.7.2
Using sprockets-rails 3.2.2
Using rails 5.2.4.4
Using blacklight 7.0.1
Using parslet 1.8.2
Using blacklight_advanced_search 7.0.0.alpha from https://github.com/projectblacklight/blacklight_advanced_search.git (at master@a936db0)
Using jquery-rails 4.3.5
Using tether-rails 1.4.0
Using blacklight_range_limit 7.0.0
Using msgpack 1.3.0
Using bootsnap 1.4.4
Using popper_js 1.14.5
Using ffi 1.11.1
Using sassc 2.0.1
Using tilt 2.0.10
Using sassc-rails 2.1.2
Using bootstrap 4.3.1
Using byebug 11.0.1
Using regexp_parser 1.6.0
Using xpath 3.2.0
Using capybara 3.27.0
Using launchy 2.4.3
Using capybara-screenshot 1.0.23
Using childprocess 1.0.1
Using rubyzip 1.3.0
Using selenium-webdriver 3.142.3
Using capybara-selenium 0.0.6
Using mime-types-data 3.2019.1009
Using mime-types 3.3
Using carrierwave 1.3.1
Using coffee-script-source 1.12.2
Using coffee-script 2.4.1
Using coffee-rails 4.2.2
Using rb-fsevent 0.10.3
Using rb-inotify 0.10.0
Using sass-listen 4.0.0
Using sass 3.7.4
Using sass-rails 5.0.7
Using chosen-rails 1.8.7
Using chronic 0.10.2
Using coderay 1.1.2
Using deep_merge 1.2.1
Using dry-core 0.4.9
Using dry-configurable 0.8.3
Using dry-equalizer 0.2.2
Using dry-initializer 3.0.1
Using dry-logic 1.0.3
Using dry-container 0.7.2
Using dry-inflector 0.2.0
Using dry-types 1.2.0
Using dry-schema 1.4.1
Using config 2.0.0
Using connection_pool 2.2.2
Using database_cleaner 1.7.0
Using orm_adapter 0.5.0
Using responders 3.0.0
Using warden 1.2.8
Using devise 4.7.1
Using devise-guests 0.7.0
Using diff-lcs 1.3
Using dotenv 2.7.4
Using dotenv-rails 2.7.4
Using down 4.8.1
Using ethon 0.12.0
Using exception_notification 4.3.0
Using factory_bot 5.0.2
Using factory_bot_rails 5.0.2
Using multipart-post 2.1.1
Using faraday 0.17.0
Using faraday_middleware 0.13.1
Using font-awesome-rails 4.7.0.5
Using foreman 0.85.0
Using json-schema 2.8.1
Using net-http-persistent 2.9.4
Using rsolr 2.2.1
Using nokogumbo 2.0.2
Using sanitize 5.2.1
Using geo_combine 0.4.0
Using geoblacklight-icons 1.3.1 from https://github.com/geoblacklight/geoblacklight-icons.git (at 84da811@84da811)
Using handlebars_assets 0.23.5
Using rack-proxy 0.6.5
Using webpacker 4.0.7
Using geoblacklight 2.2.1
Using mini_magick 4.9.5
Using ruby-vips 2.0.16
Using image_processing 1.9.3
Using statesman 3.4.1
Using geoblacklight_sidecar_images 0.6.2
Using temple 0.8.1
Using haml 5.1.1
Using letter_opener 1.7.0
Using ruby_dep 1.5.0
Using listen 3.1.5
Using m 1.5.1
Using minitest-ci 3.4.0
Using ruby-progressbar 1.10.1
Using minitest-reporters 1.3.6
Using mysql2 0.5.2
Using typhoeus 0.8.0
Using pointless_feedback 4.0.6
Using puma 4.3.5
Using rack-cors 1.0.5
Using rack-protection 2.0.5
Using redis 4.1.2
Using retriable 3.1.2
Using rspec-support 3.8.2
Using rspec-core 3.8.2
Using rspec-expectations 3.8.4
Using rspec-mocks 3.8.1
Using rspec-rails 3.8.2
Using sidekiq 5.1.3
Using sidekiq-failures 1.0.0
Using sitemap_generator 6.0.2
Using solr_wrapper 2.1.0 from https://github.com/cbeer/solr_wrapper.git (at master@cd17e49)
Using spring 2.1.0
Using sqlite3 1.3.13
Using turbolinks-source 5.2.0
Using turbolinks 5.2.0
Using twitter-typeahead-rails 0.11.1.pre.corejavascript
Using uglifier 4.1.20
Using web-console 3.7.0
Using webdrivers 4.1.1
Using whenever 0.9.7
Bundle updated!

$ bundle install
Using rake 12.3.3
Using concurrent-ruby 1.1.7
Using i18n 1.8.5
Using minitest 5.14.2
Using thread_safe 0.3.6
Using tzinfo 1.2.7
Using activesupport 5.2.4.4
Using builder 3.2.4
Using erubi 1.9.0
Using mini_portile2 2.4.0
Using nokogiri 1.10.10
Using rails-dom-testing 2.0.3
Using crass 1.0.6
Using loofah 2.7.0
Using rails-html-sanitizer 1.3.0
Using actionview 5.2.4.4
Using rack 2.2.3
Using rack-test 1.1.0
Using actionpack 5.2.4.4
Using nio4r 2.5.4
Using websocket-extensions 0.1.5
Using websocket-driver 0.7.3
Using actioncable 5.2.4.4
Using globalid 0.4.2
Using activejob 5.2.4.4
Using mini_mime 1.0.2
Using mail 2.7.1
Using actionmailer 5.2.4.4
Using activemodel 5.2.4.4
Using arel 9.0.0
Using activerecord 5.2.4.4
Using mimemagic 0.3.10
Using marcel 0.3.3
Using activestorage 5.2.4.4
Using public_suffix 3.1.1
Using addressable 2.5.2
Using ansi 1.5.0
Using execjs 2.7.0
Using autoprefixer-rails 9.6.1
Using awesome_print 1.8.0
Using bcrypt 3.1.13
Using bindex 0.8.1
Using deprecation 1.0.0
Using jbuilder 2.9.1
Using kaminari-core 1.2.1
Using kaminari-actionview 1.2.1
Using kaminari-activerecord 1.2.1
Using kaminari 1.2.1
Using bundler 2.1.4
Using method_source 1.0.0
Using thor 0.19.4
Using railties 5.2.4.4
Using sprockets 3.7.2
Using sprockets-rails 3.2.2
Using rails 5.2.4.4
Using blacklight 7.0.1
Using parslet 1.8.2
Using blacklight_advanced_search 7.0.0.alpha from https://github.com/projectblacklight/blacklight_advanced_search.git (at master@a936db0)
Using jquery-rails 4.3.5
Using tether-rails 1.4.0
Using blacklight_range_limit 7.0.0
Using msgpack 1.3.0
Using bootsnap 1.4.4
Using popper_js 1.14.5
Using ffi 1.11.1
Using sassc 2.0.1
Using tilt 2.0.10
Using sassc-rails 2.1.2
Using bootstrap 4.3.1
Using byebug 11.0.1
Using regexp_parser 1.6.0
Using xpath 3.2.0
Using capybara 3.27.0
Using launchy 2.4.3
Using capybara-screenshot 1.0.23
Using childprocess 1.0.1
Using rubyzip 1.3.0
Using selenium-webdriver 3.142.3
Using capybara-selenium 0.0.6
Using mime-types-data 3.2019.1009
Using mime-types 3.3
Using carrierwave 1.3.1
Using coffee-script-source 1.12.2
Using coffee-script 2.4.1
Using coffee-rails 4.2.2
Using rb-fsevent 0.10.3
Using rb-inotify 0.10.0
Using sass-listen 4.0.0
Using sass 3.7.4
Using sass-rails 5.0.7
Using chosen-rails 1.8.7
Using chronic 0.10.2
Using coderay 1.1.2
Using deep_merge 1.2.1
Using dry-core 0.4.9
Using dry-configurable 0.8.3
Using dry-equalizer 0.2.2
Using dry-initializer 3.0.1
Using dry-logic 1.0.3
Using dry-container 0.7.2
Using dry-inflector 0.2.0
Using dry-types 1.2.0
Using dry-schema 1.4.1
Using config 2.0.0
Using connection_pool 2.2.2
Using database_cleaner 1.7.0
Using orm_adapter 0.5.0
Using responders 3.0.0
Using warden 1.2.8
Using devise 4.7.1
Using devise-guests 0.7.0
Using diff-lcs 1.3
Using dotenv 2.7.4
Using dotenv-rails 2.7.4
Using down 4.8.1
Using ethon 0.12.0
Using exception_notification 4.3.0
Using factory_bot 5.0.2
Using factory_bot_rails 5.0.2
Using multipart-post 2.1.1
Using faraday 0.17.0
Using faraday_middleware 0.13.1
Using font-awesome-rails 4.7.0.5
Using foreman 0.85.0
Using json-schema 2.8.1
Using net-http-persistent 2.9.4
Using rsolr 2.2.1
Using nokogumbo 2.0.2
Using sanitize 5.2.1
Using geo_combine 0.4.0
Using geoblacklight-icons 1.3.1 from https://github.com/geoblacklight/geoblacklight-icons.git (at 84da811@84da811)
Using handlebars_assets 0.23.5
Using rack-proxy 0.6.5
Using webpacker 4.0.7
Using geoblacklight 2.2.1
Using mini_magick 4.9.5
Using ruby-vips 2.0.16
Using image_processing 1.9.3
Using statesman 3.4.1
Using geoblacklight_sidecar_images 0.6.2
Using temple 0.8.1
Using haml 5.1.1
Using letter_opener 1.7.0
Using ruby_dep 1.5.0
Using listen 3.1.5
Using m 1.5.1
Using minitest-ci 3.4.0
Using ruby-progressbar 1.10.1
Using minitest-reporters 1.3.6
Using mysql2 0.5.2
Using typhoeus 0.8.0
Using pointless_feedback 4.0.6
Using puma 4.3.5
Using rack-cors 1.0.5
Using rack-protection 2.0.5
Using redis 4.1.2
Using retriable 3.1.2
Using rspec-support 3.8.2
Using rspec-core 3.8.2
Using rspec-expectations 3.8.4
Using rspec-mocks 3.8.1
Using rspec-rails 3.8.2
Using sidekiq 5.1.3
Using sidekiq-failures 1.0.0
Using sitemap_generator 6.0.2
Using solr_wrapper 2.1.0 from https://github.com/cbeer/solr_wrapper.git (at master@cd17e49)
Using spring 2.1.0
Using sqlite3 1.3.13
Using turbolinks-source 5.2.0
Using turbolinks 5.2.0
Using twitter-typeahead-rails 0.11.1.pre.corejavascript
Using uglifier 4.1.20
Using web-console 3.7.0
Using webdrivers 4.1.1
Using whenever 0.9.7
Bundle complete! 62 Gemfile dependencies, 175 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.

$ bundle exec rake geoportal:server
Loading configuration from /Users/zing/Desktop/RA/GitHub/geoportal/.solr_wrapper
rake aborted!
NoMethodError: undefined method `[]' for nil:NilClass
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper/configuration.rb:191:in `fetch_latest_version'
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper/configuration.rb:89:in `version'
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper/instance.rb:292:in `extracted?'
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper/instance.rb:256:in `extract'
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper/instance.rb:247:in `extract_and_configure'
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper/instance.rb:62:in `wrap'
/Users/zing/.rvm/gems/ruby-2.6.1/bundler/gems/solr_wrapper-cd17e4908825/lib/solr_wrapper.rb:43:in `wrap'
/Users/zing/Desktop/RA/GitHub/geoportal/lib/tasks/geoportal.rake:27:in `block (2 levels) in <main>'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
/Users/zing/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:22:in `eval'
/Users/zing/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:22:in `<main>'
Tasks: TOP => geoportal:server
(See full trace by running task with --trace)
```
It looks like there is something wrong with the `.solr_wrapper` stuff, but I not sure whether it came from the `solr_wrapper GitHub repo` or the local `.solr_wrapper file`. I will be really grateful if you are able to offer some help. Many thanks😅


-Gene

**Comment by ewlarson** [comment]
Hi @Ziiiiing / I noticed the same issue with the solr_wrapper gem while working on the GEOMG project this week, but I just assumed it was something related to my local development computer.

I resolved the problem there by bumping the solr_wrapper dependency up to its latest release:
https://github.com/BTAA-Geospatial-Data-Project/geomg/commit/882715de2115402dc2331f72ac627f05c1e62c61

Try changing your local Geoportal Gemfile to do the same:
`gem 'solr_wrapper', '~> 3.1'`

That should resolve whatever issue is happening there.

**Comment by Ziiiiing** [comment]
Hi @ewlarson, thanks for your help.

I have tried to change the **Gemfile** with  `gem 'solr_wrapper', '~> 3.1'`. However, it still raises `rake aborted!` but with a different reason `Not started yet`  rather than `NoMethodError: undefined method `[]' for nil:NilClass` shown before. Here's what the error looks like😞

```console
$ bundle exec rake geoportal:server
Loading configuration from /Users/zing/Desktop/RA/GitHub/geoportal/.solr_wrapper
solr-7.7.1.zip: |=======================================| 100% (Time: 00:25:22 )
rake aborted!
Not started yet
/Users/zing/.rvm/gems/ruby-2.6.1/gems/solr_wrapper-3.1.1/lib/solr_wrapper/instance.rb:152:in `block in create'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/retriable-3.1.2/lib/retriable.rb:61:in `block in retriable'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/retriable-3.1.2/lib/retriable.rb:56:in `times'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/retriable-3.1.2/lib/retriable.rb:56:in `retriable'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/solr_wrapper-3.1.1/lib/solr_wrapper/instance.rb:151:in `create'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/solr_wrapper-3.1.1/lib/solr_wrapper/instance.rb:215:in `with_collection'
/Users/zing/Desktop/RA/GitHub/geoportal/lib/tasks/geoportal.rake:28:in `block (3 levels) in <main>'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/solr_wrapper-3.1.1/lib/solr_wrapper/instance.rb:64:in `wrap'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/solr_wrapper-3.1.1/lib/solr_wrapper.rb:43:in `wrap'
/Users/zing/Desktop/RA/GitHub/geoportal/lib/tasks/geoportal.rake:27:in `block (2 levels) in <main>'
/Users/zing/.rvm/gems/ruby-2.6.1/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
/Users/zing/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:22:in `eval'
/Users/zing/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:22:in `<main>'
Tasks: TOP => geoportal:server
(See full trace by running task with --trace)
```

**Comment by ewlarson** [comment]
So sorry for the trouble @Ziiiiing. I've not seen that error before. I wonder if we don't have enough documentation about all the dependencies to run the Geoportal here. Do you have Java 1.8+ installed? 

Have you ever run GeoBlacklight on your local machine? Their documentation is much better. Try and run through this tutorial:
https://geoblacklight.org/tutorial/2015/02/09/setting-up-your-environment.html

If that spins up correctly for you, you will be able to get the B1G Geoportal up and running.

**Comment by mberkowski** [comment]
I will try to make time today to spin one up from scratch too, since I haven't in many months. It's likely I could spot a doc issue then.

**Comment by mberkowski** [comment]
Hmm, I tested by intentionally breaking my own java install but that caused a different error state in solr_wrapper, not the one @Ziiiiing detailed. Some things to try:

Is a solr already running, causing solr_wrapper to get confused? Try

```
$ ps aux | grep '[s]olr'
```
which if solr is running somewhere, will give you a bunch of output like:

```
mjb      2711516  8.9  1.4 4136356 229072 pts/3  Sl   11:13 0:07 java -server -Xms512m -Xmx512m -XX:NewRatio=3 -XX:SurvivorRatio=4 -XX:TargetSurvivorRatio=90 -XX:MaxTenuringThreshold=8 -XX:+UseConcMarkSweepGC -XX:ConcGCThreads=4 -XX:ParallelGCThreads=4 -XX:+CMSScavengeBeforeRemark -XX:PretenureSizeThreshold=64m -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=50 -XX:CMSMaxAbortablePrecleanTime=6000 -XX:+CMSParallelRemarkEnabled -XX:+ParallelRefProcEnabled -XX:-OmitStackTraceInFastThrow -Xlog:gc*:file=/....
...
...
```

If it does show output like the above, try to kill the process by its number. In mine above, the second value `2711516` is the process id I want to eliminate. After this, try starting `geoportal:server` again.

```
$ kill -9 2711516
```

If you did not find any Solr already runing, I think the next thing I'd probably do is delete everything in the geoportal project's `tmp/` folder

Then before running the whole `geoportal:server` bundle, maybe try starting solr alone with

```
$ bundle exec rake solr:start
```

If that works, stop it with `bundle exec rake solr:stop` then try `geoportal:server` again.

**Comment by Ziiiiing** [comment]
Hi @ewlarson  @mberkowski ,

I guess I find out the solution for starting the Solr on my local computer in the end. Besides updating the local **Gemfile** with `gem 'solr_wrapper', '~> 3.1'`, we should also check the version of Java before running the application. The Java has to be **version 1.8** ( or called **Java 8**) only so that the Solr is able to start successfully. The default Java version on my laptop was 1.15 and that's why I failed to start it.

Here are some details I used for solving this issue. Hope it can help.

First, check your default (or let's say current) Java version before running the application with:
```console
$ java -version
java version "15.0.2" 2021-01-19
```
If you got the version rather than 1.8, try to run `/usr/libexec/java_home -V` to check all Java versions installed on your computer, and the output might look like this:
```console
$ /usr/libexec/java_home -V
Matching Java Virtual Machines (2):
    15.0.2 (x86_64) "Oracle Corporation" - "Java SE 15.0.2" /Library/Java/JavaVirtualMachines/jdk-15.0.2.jdk/Contents/Home
    1.8.271.09 (x86_64) "Oracle Corporation" - "Java" /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home
/Library/Java/JavaVirtualMachines/jdk-15.0.2.jdk/Contents/Home
```
Swith the default version to **Java 8** (`1.8.271.09` on my laptop) with:
```console
export JAVA_HOME=`/usr/libexec/java_home -v 1.8.271.09`
```
Then you will get the Java 8 in the end, and you are good to start the Solr and run the application:
```console
$ java -version
java version "1.8.0_271"
```

**Reminder: If you do not have Java 8 installed on your local computer, try to follow [this website](https://devqa.io/brew-install-java/) to install Java 8 first.**
